PWGCF/FemtoWorld/Tasks/femtoWorldEfficiencyTaskDe.cxx

```diff
@@ -0,0 +1,522 @@
+
+  void processReco(const CollisionsEvSel::iterator& collision,
+                   soa::Filtered<soa::Join<TracksPID, aod::TrackSelection>> const& tracks /*, aod::BCsWithTimestamps const&*/)
+  {
+    if (ConfIsRun3) {        //
+      if (!collision.sel8()) //  CHECK
+        return;              //
+    }
+
+    // auto bc = collision.bc_as<aod::BCsWithTimestamps>(); /// adding timestamp to access magnetic field later
+    //  Default event selection
+    // colCuts.printCuts();
+    // if (!colCuts.isSelected(collision)) //
+    // return;                           //  CHECK -- DATA BLOCKAGE
+    // colCuts.fillQA(collision);          //
+
+    // Loop over tracks
+    for (auto& track : tracks) {
+      // Tracks are already filtered by the pre-filters
+
+      // track.hasTOF() - to check if TOF info available
+
+      // cuts on tracks:
+      // if (!ConfPIDnoTOF) {                                                      //
+      // if (track.pt() > tofPtCut && !track.hasTOF())                           //  CHECK
+      // continue; // if no TOF information above tofPtCut reject such track   //
+      //}
+
+      registryGlobal.fill(HIST("crossedRows"), track.tpcNClsCrossedRows());
+      registryGlobal.fill(HIST("RowsOverClustersTPC"), track.tpcCrossedRowsOverFindableCls());
+      registryGlobal.fill(HIST("chi2TPC"), track.tpcChi2NCl());
+      registryGlobal.fill(HIST("chi2ITS"), track.itsChi2NCl());
+      registryGlobal.fill(HIST("dcaZ"), track.dcaZ());
+      registryGlobal.fill(HIST("dcaXY"), track.dcaXY());
+      registryGlobal.fill(HIST("clustersITS"), track.itsNCls());
+      registryGlobal.fill(HIST("pt"), track.pt());
+      registryGlobal.fill(HIST("eta"), track.eta());
+
+      // no PID histograms
+      if (track.sign() > 0) {
+        registryQAtrack.fill(HIST("after/all/plus/etaphi"), track.eta(), track.phi());
+        registryQAtrack.fill(HIST("after/all/plus/pt"), track.pt());
+        registryQAtrack.fill(HIST("after/all/plus/eta"), track.eta());
+        registryQAtrack.fill(HIST("after/all/plus/phi"), track.phi());
+        registryQAtrack.fill(HIST("after/all/plus/DCAxy"), track.pt(), track.dcaXY());
+        registryQAtrack.fill(HIST("after/all/plus/DCAz"), track.pt(), track.dcaZ());
+      }
+      if (track.sign() < 0) {
+        registryQAtrack.fill(HIST("after/all/minus/etaphi"), track.eta(), track.phi());
+        registryQAtrack.fill(HIST("after/all/minus/pt"), track.pt());
+        registryQAtrack.fill(HIST("after/all/minus/eta"), track.eta());
+        registryQAtrack.fill(HIST("after/all/minus/phi"), track.phi());
+        registryQAtrack.fill(HIST("after/all/minus/DCAxy"), track.pt(), track.dcaXY());
+        registryQAtrack.fill(HIST("after/all/minus/DCAz"), track.pt(), track.dcaZ());
+      }
+
+      // Add PID selection criteria here
+      if (IsNSigmaAccept(std::abs(track.tpcNSigmaPi()), std::abs(track.tofNSigmaPi()), track.pt())) {
```

Comment:
If IsNSigmaAccept is templated (for the future when MC reco were used) it is better to pass to it the track object to avoid the storage of the track parameters in advance